pkgs/development/interpreters/python/hooks/setuptools-rust-hook.sh

```diff
@@ -0,0 +1,18 @@
+echo "Sourcing setuptools-rust-hook"
+
+setuptoolsRustSetup() {
+    # This can work only if rustPlatform.cargoSetupHook is also included
+    if ! command -v cargoSetupPostPatchHook >/dev/null; then
+        echo "ERROR: setuptools-rust has to be used alongside with rustPlatform.cargoSetupHook!"
+        exit 1
+    fi
+
+    export PYO3_CROSS_LIB_DIR="@pyLibDir@"
+    export CARGO_BUILD_TARGET=@cargoBuildTarget@
+    # TODO theoretically setting linker should not be required because it is
+    # already set in pkgs/build-support/rust/hooks/default.nix but build fails
+    # on missing linker without this.
+    export CARGO_TARGET_@cargoLinkerVar@_LINKER=@targetLinker@
```

Comment:
I don't understand how not to use `rustPlatform`. How can I access the `cargoSetupHook` unless I am using `buildRustPackage` or through `rustPlatform`? I can't use `buildRustPackage` because this is a Python hook primarily.

Reply:
> I don't understand how not to use `rustPlatform`.

Let me be more explicit:

> If you need this, it means splicing isn't working right. That's usually caused by referencing ~things~ **rustc and cargo** via `rustPlatform`. Don't do that. Use my next suggestion instead.

Reply:
It appears that you have applied this suggestion.